yt_dlp/extractor/cnn.py

```diff
@@ -1,146 +1,225 @@
+            'info_dict': {
+                'id': 'me19d548fdd54df0924087039283128ef473ab397d',
+                'ext': 'mp4',
+                'title': '\'I\'m not finished\': Harris interview with Fox News gets heated',
+                'display_id': 'kamala-harris-fox-news-interview-ebof-digvid',
+                'description': 'md5:e7dd3d1a04df916062230b60ca419a0a',
+                'thumbnail': 'https://media.cnn.com/api/v1/images/stellar/prod/harris-20241016234916617.jpg?c=original',
+                'duration': 173.0,
+                'timestamp': 1729122182.079,
+                'upload_date': '20241016',
+                'modified_timestamp': 1729194706.355,
+                'modified_date': '20241017',
+            },
+            'params': {'format': 'direct'},
+        }, {
+            'md5': '11604ab4af83b650826753f1ccb8ecff',
+            'info_dict': {
+                'id': 'med04507d8ca3da827001f63d22af321ec29c7d97b',
+                'ext': 'mp4',
+                'title': '\'Wise\': Buttigieg on Harris\' handling of interview question about gender transition surgery',
+                'display_id': 'pete-buttigieg-harris-fox-newssrc-digvid',
+                'description': 'md5:602a8a7e853ed5e574acd3159428c98e',
+                'thumbnail': 'https://media.cnn.com/api/v1/images/stellar/prod/buttigieg-20241017040412074.jpg?c=original',
+                'duration': 145.0,
+                'timestamp': 1729137765.846,
+                'upload_date': '20241017',
+                'modified_timestamp': 1729138184.891,
+                'modified_date': '20241017',
+            },
+            'params': {'format': 'direct'},
+        }],
+    }]
 
     def _real_extract(self, url):
-        sub_domain, path, page_title = self._match_valid_url(url).groups()
-        if sub_domain not in ('money', 'edition'):
-            sub_domain = 'edition'
-        config = self._CONFIG[sub_domain]
-        return self._extract_cvp_info(
-            config['data_src'] % path, page_title, {
-                'default': {
-                    'media_src': config['media_src'],
-                },
-                'f4m': {
-                    'host': 'cnn-vh.akamaihd.net',
-                },
-            })
+        display_id = self._match_valid_url(url).group('display_id')
+        webpage = self._download_webpage(url, display_id)
+        app_id = traverse_obj(
+            self._search_json(r'window\.env\s*=', webpage, 'window env', display_id, default={}),
+            ('TOP_AUTH_SERVICE_APP_ID', {str}))
 
+        entries = []
+        for player_data in traverse_obj(webpage, (
+                {find_elements(tag='div', attr='data-component-name', value='video-player', html=True)},
+                ..., {extract_attributes}, all, lambda _, v: v['data-media-id'] and v['data-video-resource-parent-uri'])):
+            media_id = player_data['data-media-id']
+            parent_uri = player_data['data-video-resource-parent-uri']
```

Comment:
not a necessary change, but maybe we want to remove the filter for `parent_uri` here, since we could still have m3u8 formats with only the `media_id`?
```suggestion
                ..., {extract_attributes}, all, lambda _, v: v['data-media-id'])):
            media_id = player_data['data-media-id']
            parent_uri = player_data.get('data-video-resource-parent-uri')
```